src/core/algorithms/order/order.cpp

```diff
@@ -0,0 +1,366 @@
+#include "order.h"
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+#include <easylogging++.h>
+
+#include "config/names_and_descriptions.h"
+#include "config/tabular_data/input_table/option.h"
+#include "dependency_checker.h"
+#include "list_lattice.h"
+#include "model/types/types.h"
+#include "order_utility.h"
+
+namespace algos::order {
+
+Order::Order() : Algorithm({}) {
+    RegisterOptions();
+    MakeOptionsAvailable({config::TableOpt.GetName()});
+}
+
+void Order::RegisterOptions() {
+    using namespace config::names;
+    using namespace config::descriptions;
+    using config::Option;
+
+    RegisterOption(config::TableOpt(&input_table_));
+}
+
+void Order::LoadDataInternal() {
+    typed_relation_ = model::ColumnLayoutTypedRelationData::CreateFrom(*input_table_, false);
+}
+
+void Order::ResetState() {}
+
+void Order::CreateSingletonSortedPartitions() {
+    std::vector<model::TypedColumnData> const& data = typed_relation_->GetColumnData();
+    std::unordered_set<unsigned long> null_rows;
+    for (unsigned int i = 0; i < data.size(); ++i) {
+        if (!model::Type::IsOrdered(data[i].GetTypeId()) && !data[i].IsMixed()) {
```

Comment:
`Mixed` is not considered `Ordered`, so if the first check fails the second will never evaluate to true